### ISSUE TITLE
Ability to use private IP address for Azure pod VM image creation

### ISSUE THREAD

**Issue by bpradipt** [confidential-containers/cloud-api-adaptor#1465]
Currently when using packer to build pod VM image on Azure, public IP is used. This creates a problem in some environments where public IP is disabled as a policy. 
This issue is to track adding support for Azure pod VM image creation using private IP. 

One option is to use an approach like AWS where explicit network details are provided for pod vm image creation.  

Here is a sample code on similar lines that I test successfully - https://github.com/bpradipt/cloud-api-adaptor/commit/728f2eebd8ce899e42f624ea91e940e79c7e8dad

**Comment by mkulke** [comment]
That would only work if packer was invoked on a private vnet (i.e. not on public github actions runners), if I understand correctly. But I don't see a problem with providing the options for non-github based builds.

**Comment by bpradipt** [comment]
@mkulke right. This is for private environments. Also it should be optional, with default being the current behaviour where packer creates a VM with public IP

**Comment by kartikjoshi21** [comment]
@bpradipt is it still required as we are slowly moving to `mkosi` based builds ?
